src/handlers/transactions.ts

```diff
@@ -0,0 +1,156 @@
+    payable: false,
+    type: 'function',
+  },
+  {
+    constant: false,
+    inputs: [{ name: '_method', type: 'string' }],
+    name: 'register',
+    outputs: [{ name: '', type: 'bool' }],
+    payable: false,
+    type: 'function',
+  },
+  { inputs: [], type: 'constructor' },
+  {
+    anonymous: false,
+    inputs: [
+      { indexed: true, name: 'creator', type: 'address' },
+      { indexed: true, name: 'signature', type: 'bytes4' },
+      { indexed: false, name: 'method', type: 'string' },
+    ],
+    name: 'Registered',
+    type: 'event',
+  },
+  {
+    anonymous: false,
+    inputs: [
+      { indexed: true, name: 'old', type: 'address' },
+      { indexed: true, name: 'current', type: 'address' },
+    ],
+    name: 'NewOwner',
+    type: 'event',
+  },
+];
+
+const parseSignatureToTitle = (signature: string) => {
+  const rawName = signature.match(/^([^)(]*)\((.*)\)([^)(]*)$/u);
+  let parsedName = '';
+
+  if (rawName) {
+    parsedName =
+      rawName[1].charAt(0).toUpperCase() +
+      rawName[1]
+        .slice(1)
+        .split(/(?=[A-Z])/u)
+        .join(' ');
+  }
+  return parsedName;
+};
+
+const fourByteApi = new RainbowFetchClient({
+  baseURL: 'https://www.4byte.directory/api/v1/signatures',
+  headers: {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json',
+  },
+  timeout: 800,
+});
+
+const timeoutPromise = new Promise((_, reject) => {
+  setTimeout(reject, 800);
+});
+
+export const getTransactionMethodName = async (
+  transaction: ZerionTransaction
+) => {
+  try {
+    const { transactionSignatures } = store.getState().data;
+    // only being used on mainnet transactions, so we can use the default web3 provider
+    const txn = await web3Provider.getTransaction(transaction.hash);
+    const bytes = txn?.data?.substring(0, 10) || '';
+    let signature = transactionSignatures[bytes] || '';
+    if (signature) return signature;
+    try {
+      const response = await fourByteApi.get(`/?hex_signature=${bytes}`);
+      const responseData: any = response?.data;
+      const bestResult = responseData?.results.sort(
+        (a: { id: number }, b: { id: number }) => (a.id < b.id ? -1 : 1)
+      )?.[0];
+      signature = bestResult.text_signature;
+      // eslint-disable-next-line no-empty
+    } catch (e) {}
+    if (!signature) {
+      try {
+        const contract = new Contract(
+          SIGNATURE_REGISTRY_ADDRESS,
+          abi,
+          web3Provider
+        );
+        signature = await Promise.race([
+          contract.entries(bytes),
+          timeoutPromise,
+        ]);
+        // eslint-disable-next-line no-empty
+      } catch (e) {}
+    }
+    const parsedSignature = parseSignatureToTitle(signature);
+    if (parsedSignature) {
+      const newTransactionSignatures = {
+        ...transactionSignatures,
+        [bytes]: parsedSignature,
+      };
+      try {
+        saveTransactionSignatures(newTransactionSignatures);
```

Comment:
Can update redux here, and then the redux logic can update localstorage

Reply:
done!

Reply:
cool!
1. can we follow the pattern we have in redux where the redux file exposes a function that does these 2 things (dispatch and save txn signatures to localstorage; so then we don't need to export random dispatch action types everywhere etc)
2. also now that we are actually using the redux state for this global signature data, could we stick it in its own redux file instead of `redux/data`? (ideally `redux/data` should be used for account-specific info (even though that isn't entirely true right now) and so we won't be "clearing" the entire signature data when we switch accounts))
Things to keep in mind when doing this: the redux loading function should be called from `useLoadGlobalData` instead of `useLoadAccountData` (you can follow the redux pattern in `redux/topMovers` for instance)